packages/core/src/libraries/hook/context-manager.ts

```diff
@@ -21,21 +23,44 @@ type DataHookMetadata = {
   ip: string;
 } & Partial<InteractionApiMetadata>;
 
-type DataHookContext = {
-  event: DataHookEvent;
+export type DataHookContext = {
   /** Data details */
   data?: unknown;
 } & Partial<ManagementApiContext> &
   Record<string, unknown>;
 
+type UserContext = {
+  /**
+   * This user will be picked with {@link userInfoSelectFields} and set to the `data` field. The
+   * original user object will be discarded.
+   *
+   * @example
+   * const context = { user: { ... } };
+   *
+   * // The actual context to send will be:
+   * { data: pick(user, ...userInfoSelectFields) }
+   */
+  user: User;
+};
+
+/**
+ * A map of data hook event to its context type for better type hinting.
+ */
+export type DataHookContextMap = {
```

Comment:
nit: Would it be better to put this type in @logto/schemas? Keep the same resource as the DataHook events definitions.

Reply:
i doubt this. it's a core-specific type and it may change time to time - unless our users are using `@logto/schemas` for their hook apis

Reply:
aligned offline